Objects/frameobject.c

```diff
@@ -826,6 +831,10 @@ frame_setlineno(PyFrameObject *f, PyObject* p_new_lineno, void *Py_UNUSED(ignore
         }
         start_stack = pop_value(start_stack);
     }
+    // Populate any NULL locals that might be needed:
+    for (int i = 0; i < f->f_frame->f_code->co_nlocalsplus; i++) {
+        maybe_bind_fast_local(f->f_frame, i);
+    }
```

Comment:
I believe this could be converted from quadratic to linear time if we wanted to allocate an array of `nlocals` bytes or bits to first flag all the relevant locals in one pass, then later bind `None` to those locals flagged.

Maybe it's not a huge issue in practice, but it would be nice to not have to scan the whole code object for each local.

On the other hand, I suspect most locals have a `LOAD_FAST` somewhere, so maybe it would be easier to just bind `None` to *any* uninitialized local when a frame jump happens.